contracts/lootbox-v1/contracts/Lootbox.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./interfaces/IUniswapV2Router02.sol";
+import "../../dollar/contracts/interfaces/IUbiquityAlgorithmicDollarManager.sol";
+import "../../dollar/contracts/interfaces/IUbiquityAlgorithmicDollar.sol";
+import "./interfaces/IWyvernExchange.sol";
+
+contract Lootbox is Ownable {
+
+  IUbiquityAlgorithmicDollarManager public dollarManager;
+  IUbiquityAlgorithmicDollar public dollarToken;
+  IWyvernExchange public wyvernExchange;
+
+  address public creditToken;
+
+  uint256 public transactionFee;
+  uint256 public cashbackPercentage;
+  address private constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
```

Comment:
Aggregator might be able to abstract the asset conversations for us. Upon cursory inspection of the transactions to Genie, they all seem to be in Ether.

Reply:
Yes, I think so too. they all seem to be in Ether in Genie no need WETH but need to investigate more detail.
I though, there is no choice without using WETH on Uniswap or Opensea